lib/extensions/virtual_total.rb

```diff
@@ -93,17 +93,33 @@ def define_virtual_aggregate_method(name, relation, method_name, column)
     def virtual_aggregate_arel(reflection, method_name, column)
       return unless reflection && reflection.macro == :has_many && !reflection.options[:through]
       lambda do |t|
+        query = if reflection.scope
+                  reflection.klass.instance_exec(nil, &reflection.scope)
+                else
+                  reflection.klass.all
+                end
+
+        # ordering will probably screw up aggregations, so clear this out from
+        # any calls
+        #
+        # only clear this out if this isn't a `:size` call as well, since doing
+        # a COUNT(*) will allow any ORDER BY to still work properly.  This is
+        # to avoid any possible edge cases by clearing out the order clause.
+        query.order_values = [] if method_name != :size
+
         foreign_table = reflection.klass.arel_table
         # need db access for the keys, so delaying all this lookup until call time
-        local_key = reflection.active_record_primary_key
+        local_key   = reflection.active_record_primary_key
         foreign_key = reflection.foreign_key
+        query       = query.where(t[local_key].eq(foreign_table[foreign_key]))
+
         arel_column = if method_name == :size
                         Arel.star.count
                       else
                         reflection.klass.arel_attribute(column).send(method_name)
                       end
-        t.grouping(foreign_table.project(arel_column)
-                                .where(t[local_key].eq(foreign_table[foreign_key])))
+
+        t.grouping(Arel::Nodes::SqlLiteral.new(query.select(arel_column).to_sql))
```

Comment:
Ugh. I am frustrated that `Arel` makes you do this. 😦 
I do recognize the pattern and 👍 😦

Reply:
can you move the `select(arel_column)` out of this line. Not sure if you want to put as part of the `arel_column` block or right after it.

You see, the `query = query.where` clause that you added makes it much easier to read. Thinking that moving the `select` out of here will make the `SqlLiteral` gook easier to read.

Reply:
> can you move the `select(arel_column)`

Sure, seems reasonable.  Will try some stuff and figure out what looks best.